### PR TITLE
LU-42 Go Http Request Timeout

### PR DESCRIPTION
**Pull request by PraneethChandraThota**

### CODE REVIEW

File: examples/basic/Basic/BasicSend.go [hunk 1]
```diff
@@ -3,9 +3,9 @@ package main
 import (
 	"fmt"
 
-	"github.com/socketlabs/socketlabs-go/examples"
-	"github.com/socketlabs/socketlabs-go/injectionapi"
-	"github.com/socketlabs/socketlabs-go/injectionapi/message"
+	"github.com/PraneethChandraThota/socketlabs-go/examples"
```
Comment (david-schrenker) [inline examples/basic/Basic/BasicSend.go]: after your tests are complete please return the imports back to the main repo

File: examples/basic/BasicComplex/BasicSendComplex.go [hunk 2]
```diff
@@ -3,9 +3,9 @@ package main
 import (
 	"fmt"
 
-	exampleconfig "github.com/socketlabs/socketlabs-go/examples"
-	"github.com/socketlabs/socketlabs-go/injectionapi"
-	"github.com/socketlabs/socketlabs-go/injectionapi/message"
+	exampleconfig "github.com/PraneethChandraThota/socketlabs-go/examples"
```
Comment (david-schrenker) [inline examples/basic/BasicComplex/BasicSendComplex.go]: after your tests are complete please return the imports back to the main repo

File: examples/basic/BasicFromHtmlFile/BasicFromHtmlFile.go [hunk 3]
```diff
@@ -4,9 +4,9 @@ import (
 	"fmt"
 	"io/ioutil"
 
-	"github.com/socketlabs/socketlabs-go/examples"
-	"github.com/socketlabs/socketlabs-go/injectionapi"
-	"github.com/socketlabs/socketlabs-go/injectionapi/message"
+	"github.com/PraneethChandraThota/socketlabs-go/examples"
```
Comment (david-schrenker) [inline examples/basic/BasicFromHtmlFile/BasicFromHtmlFile.go]: after your tests are complete please return the imports back to the main repo

File: examples/basic/BasicSendWithProxy/BasicSendWithProxy.go [hunk 4]
```diff
@@ -3,15 +3,16 @@ package main
 import (
 	"fmt"
 
-	"github.com/socketlabs/socketlabs-go/examples"
-	"github.com/socketlabs/socketlabs-go/injectionapi"
-	"github.com/socketlabs/socketlabs-go/injectionapi/message"
+	"github.com/PraneethChandraThota/socketlabs-go/examples"
```
Comment (david-schrenker) [inline examples/basic/BasicSendWithProxy/BasicSendWithProxy.go]: after your tests are complete please return the imports back to the main repo

File: examples/basic/BasicWithAmpBody/BasicWithAmpBody.go [hunk 5]
```diff
@@ -3,9 +3,9 @@ package main
 import (
 	"fmt"
 
-	exampleconfig "github.com/socketlabs/socketlabs-go/examples"
-	"github.com/socketlabs/socketlabs-go/injectionapi"
-	"github.com/socketlabs/socketlabs-go/injectionapi/message"
+	exampleconfig "github.com/PraneethChandraThota/socketlabs-go/examples"
```
Comment (david-schrenker) [inline examples/basic/BasicWithAmpBody/BasicWithAmpBody.go]: after your tests are complete please return the imports back to the main repo

File: examples/basic/BasicWithApiTemplate/BasicWithApiTemplate.go [hunk 6]
```diff
@@ -3,9 +3,9 @@ package main
 import (
 	"fmt"
 
-	"github.com/socketlabs/socketlabs-go/examples"
-	"github.com/socketlabs/socketlabs-go/injectionapi"
-	"github.com/socketlabs/socketlabs-go/injectionapi/message"
+	"github.com/PraneethChandraThota/socketlabs-go/examples"
```
Comment (david-schrenker) [inline examples/basic/BasicWithApiTemplate/BasicWithApiTemplate.go]: after your tests are complete please return the imports back to the main repo

File: examples/basic/BasicWithAsciiCharset/BasicWithAsciiCharset.go [hunk 7]
```diff
@@ -3,9 +3,9 @@ package main
 import (
 	"fmt"
 
-	"github.com/socketlabs/socketlabs-go/examples"
-	"github.com/socketlabs/socketlabs-go/injectionapi"
-	"github.com/socketlabs/socketlabs-go/injectionapi/message"
+	"github.com/PraneethChandraThota/socketlabs-go/examples"
```
Comment (david-schrenker) [inline examples/basic/BasicWithAsciiCharset/BasicWithAsciiCharset.go]: after your tests are complete please return the imports back to the main repo

File: examples/basic/BasicWithAttachment/BasicWithAttachment.go [hunk 8]
```diff
@@ -3,9 +3,9 @@ package main
 import (
 	"fmt"
 
-	"github.com/socketlabs/socketlabs-go/examples"
-	"github.com/socketlabs/socketlabs-go/injectionapi"
-	"github.com/socketlabs/socketlabs-go/injectionapi/message"
+	"github.com/PraneethChandraThota/socketlabs-go/examples"
```
Comment (david-schrenker) [inline examples/basic/BasicWithAttachment/BasicWithAttachment.go]: after your tests are complete please return the imports back to the main repo

File: examples/basic/BasicWithCustomHeaders/BasicWithCustomHeaders.go [hunk 9]
```diff
@@ -3,9 +3,9 @@ package main
 import (
 	"fmt"
 
-	"github.com/socketlabs/socketlabs-go/examples"
-	"github.com/socketlabs/socketlabs-go/injectionapi"
-	"github.com/socketlabs/socketlabs-go/injectionapi/message"
+	"github.com/PraneethChandraThota/socketlabs-go/examples"
```
Comment (david-schrenker) [inline examples/basic/BasicWithCustomHeaders/BasicWithCustomHeaders.go]: after your tests are complete please return the imports back to the main repo

File: examples/basic/BasicWithEmbeddedImage/BasicWithEmbeddedImage.go [hunk 10]
```diff
@@ -3,9 +3,9 @@ package main
 import (
 	"fmt"
 
-	"github.com/socketlabs/socketlabs-go/examples"
-	"github.com/socketlabs/socketlabs-go/injectionapi"
-	"github.com/socketlabs/socketlabs-go/injectionapi/message"
+	"github.com/PraneethChandraThota/socketlabs-go/examples"
```
Comment (david-schrenker) [inline examples/basic/BasicWithEmbeddedImage/BasicWithEmbeddedImage.go]: after your tests are complete please return the imports back to the main repo

File: examples/basic/Invalid/BasicWithInvalidAttachment/BasicWithInvalidAttachment.go [hunk 11]
```diff
@@ -3,9 +3,9 @@ package main
 import (
 	"fmt"
 
-	"github.com/socketlabs/socketlabs-go/examples"
-	"github.com/socketlabs/socketlabs-go/injectionapi"
-	"github.com/socketlabs/socketlabs-go/injectionapi/message"
+	"github.com/PraneethChandraThota/socketlabs-go/examples"
```
Comment (david-schrenker) [inline examples/basic/Invalid/BasicWithInvalidAttachment/BasicWithInvalidAttachment.go]: after your tests are complete please return the imports back to the main repo

File: examples/basic/Invalid/BasicWithInvalidFrom/BasicWithInvalidFrom.go [hunk 12]
```diff
@@ -3,9 +3,9 @@ package main
 import (
 	"fmt"
 
-	"github.com/socketlabs/socketlabs-go/examples"
-	"github.com/socketlabs/socketlabs-go/injectionapi"
-	"github.com/socketlabs/socketlabs-go/injectionapi/message"
+	"github.com/PraneethChandraThota/socketlabs-go/examples"
```
Comment (david-schrenker) [inline examples/basic/Invalid/BasicWithInvalidFrom/BasicWithInvalidFrom.go]: after your tests are complete please return the imports back to the main repo

File: examples/basic/Invalid/BasicWithInvalidRecipient/BasicWithInvalidRecipient.go [hunk 13]
```diff
@@ -4,9 +4,9 @@ import (
 	"fmt"
 	"strconv"
 
-	"github.com/socketlabs/socketlabs-go/examples"
-	"github.com/socketlabs/socketlabs-go/injectionapi"
-	"github.com/socketlabs/socketlabs-go/injectionapi/message"
+	"github.com/PraneethChandraThota/socketlabs-go/examples"
```
Comment (david-schrenker) [inline examples/basic/Invalid/BasicWithInvalidRecipient/BasicWithInvalidRecipient.go]: after your tests are complete please return the imports back to the main repo

File: examples/bulk/Bulk/bulk.go [hunk 14]
```diff
@@ -3,9 +3,9 @@ package main
 import (
 	"fmt"
 
-	"github.com/socketlabs/socketlabs-go/examples"
-	"github.com/socketlabs/socketlabs-go/injectionapi"
-	"github.com/socketlabs/socketlabs-go/injectionapi/message"
+	"github.com/PraneethChandraThota/socketlabs-go/examples"
```
Comment (david-schrenker) [inline examples/bulk/Bulk/bulk.go]: after your tests are complete please return the imports back to the main repo

File: examples/bulk/BulkComplex/BulkComplex.go [hunk 15]
```diff
@@ -4,9 +4,9 @@ import (
 	"bytes"
 	"fmt"
 
-	exampleconfig "github.com/socketlabs/socketlabs-go/examples"
-	"github.com/socketlabs/socketlabs-go/injectionapi"
-	"github.com/socketlabs/socketlabs-go/injectionapi/message"
+	exampleconfig "github.com/PraneethChandraThota/socketlabs-go/examples"
```
Comment (david-schrenker) [inline examples/bulk/BulkComplex/BulkComplex.go]: after your tests are complete please return the imports back to the main repo

File: examples/bulk/BulkFromDataSourceWithMerge/BulkFromDataSourceWithMerge.go [hunk 16]
```diff
@@ -3,10 +3,10 @@ package main
 import (
 	"fmt"
 
-	"github.com/socketlabs/socketlabs-go/examples"
-	"github.com/socketlabs/socketlabs-go/examples/Integration/Repository"
-	"github.com/socketlabs/socketlabs-go/injectionapi"
-	"github.com/socketlabs/socketlabs-go/injectionapi/message"
+	"github.com/PraneethChandraThota/socketlabs-go/examples"
```
Comment (david-schrenker) [inline examples/bulk/BulkFromDataSourceWithMerge/BulkFromDataSourceWithMerge.go]: after your tests are complete please return the imports back to the main repo

File: examples/bulk/BulkWithAmpBody/BulkWithAmpBody.go [hunk 17]
```diff
@@ -3,9 +3,9 @@ package main
 import (
 	"fmt"
 
-	exampleconfig "github.com/socketlabs/socketlabs-go/examples"
-	"github.com/socketlabs/socketlabs-go/injectionapi"
-	"github.com/socketlabs/socketlabs-go/injectionapi/message"
+	exampleconfig "github.com/PraneethChandraThota/socketlabs-go/examples"
```
Comment (david-schrenker) [inline examples/bulk/BulkWithAmpBody/BulkWithAmpBody.go]: after your tests are complete please return the imports back to the main repo

File: examples/bulk/BulkWithAsciiCharsetMergeData/BulkWithAsciiCharsetMergeData.go [hunk 18]
```diff
@@ -3,9 +3,9 @@ package main
 import (
 	"fmt"
 
-	"github.com/socketlabs/socketlabs-go/examples"
-	"github.com/socketlabs/socketlabs-go/injectionapi"
-	"github.com/socketlabs/socketlabs-go/injectionapi/message"
+	"github.com/PraneethChandraThota/socketlabs-go/examples"
```
Comment (david-schrenker) [inline examples/bulk/BulkWithAsciiCharsetMergeData/BulkWithAsciiCharsetMergeData.go]: after your tests are complete please return the imports back to the main repo

File: examples/bulk/BulkWithMergeData/BulkWithMergeData.go [hunk 19]
```diff
@@ -3,9 +3,9 @@ package main
 import (
 	"fmt"
 
-	"github.com/socketlabs/socketlabs-go/examples"
-	"github.com/socketlabs/socketlabs-go/injectionapi"
-	"github.com/socketlabs/socketlabs-go/injectionapi/message"
+	"github.com/PraneethChandraThota/socketlabs-go/examples"
```
Comment (david-schrenker) [inline examples/bulk/BulkWithMergeData/BulkWithMergeData.go]: after your tests are complete please return the imports back to the main repo

File: examples/exampleconfig.go [hunk 20]
```diff
@@ -12,7 +12,7 @@ func ServerId() int {
 	//if environment variable does not exist, crash
 	if envVar == "" {
 		panic("Missing SOCKETLABS_SERVER_ID environment variable. This should be the integer value of your Socketlabs server id. " +
-			"See https://github.com/socketlabs/socketlabs-go/blob/master/README.md#obtaining-your-api-key-and-socketlabs-serverid-number")
+			"See https://github.com/PraneethChandraThota/socketlabs-go/blob/master/README.md#obtaining-your-api-key-and-socketlabs-serverid-number")
```
Comment (david-schrenker) [inline examples/exampleconfig.go]: after your tests are complete please return the imports back to the main repo

File: examples/exampleconfig.go [hunk 21]
```diff
@@ -30,7 +30,7 @@ func APIKey() string {
 	//if environment variable does not exist, crash
 	if envVar == "" {
 		panic("Missing SOCKETLABS_INJECTION_API_KEY environment variable. This should be the string value of your Socketlabs injection api key. " +
-			"See https://github.com/socketlabs/socketlabs-go/blob/master/README.md#obtaining-your-api-key-and-socketlabs-serverid-number")
+			"See https://github.com/PraneethChandraThota/socketlabs-go/blob/master/README.md#obtaining-your-api-key-and-socketlabs-serverid-number")
```
Comment (david-schrenker) [inline examples/exampleconfig.go]: after your tests are complete please return the imports back to the main repo

File: injectionapi/core/injectionrequestfactory.go [hunk 22]
```diff
@@ -1,8 +1,8 @@
 package core
 
 import (
-	"github.com/socketlabs/socketlabs-go/injectionapi/core/serialization"
-	"github.com/socketlabs/socketlabs-go/injectionapi/message"
+	"github.com/PraneethChandraThota/socketlabs-go/injectionapi/core/serialization"
+	"github.com/PraneethChandraThota/socketlabs-go/injectionapi/message"
```
Comment (david-schrenker) [inline injectionapi/core/injectionrequestfactory.go]: after your tests are complete please return the imports back to the main repo

File: injectionapi/sendvalidator.go [hunk 23]
```diff
@@ -4,7 +4,7 @@ import (
 	"strconv"
 	"strings"
 
-	"github.com/socketlabs/socketlabs-go/injectionapi/message"
+	"github.com/PraneethChandraThota/socketlabs-go/injectionapi/message"
```
Comment (david-schrenker) [inline injectionapi/sendvalidator.go]: after your tests are complete please return the imports back to the main repo

File: injectionapi/socketlabsclient.go [hunk 24]
```diff
@@ -6,10 +6,11 @@ import (
 	"net/http"
 	"net/url"
 	"runtime"
+	"time"
 
-	"github.com/socketlabs/socketlabs-go/injectionapi/core"
-	"github.com/socketlabs/socketlabs-go/injectionapi/core/serialization"
-	"github.com/socketlabs/socketlabs-go/injectionapi/message"
+	"github.com/PraneethChandraThota/socketlabs-go/injectionapi/core"
```
Comment (david-schrenker) [inline injectionapi/socketlabsclient.go]: after your tests are complete please return the imports back to the main repo

File: injectionapi/socketlabsclient.go [hunk 25]
```diff
@@ -47,6 +55,7 @@ func CreateClient(serverID int, apiKey string) ISocketlabsClient {
 		ServerID:    serverID,
 		APIKey:      apiKey,
 		EndpointURL: endpointURL,
+		RequestTimeout: 120,
```
Comment (david-schrenker) [inline injectionapi/socketlabsclient.go]: Set the default value as a const like endpointURL on line 16

File: injectionapi/socketlabsclient.go [hunk 26]
```diff
@@ -57,6 +66,7 @@ func CreateClientWithProxy(serverID int, apiKey string, proxyURL string) ISocket
 		APIKey:      apiKey,
 		EndpointURL: endpointURL,
 		ProxyURL:    proxyURL,
+		RequestTimeout: 120,
```
Comment (david-schrenker) [inline injectionapi/socketlabsclient.go]: Set the default value as a const like endpointURL on line 16

**Review by david-schrenker — CHANGES_REQUESTED** [review]
please see my comments.

File: examples/basic/BasicComplex/BasicSendComplex.go (hunk 2 above)
Comment (PraneethChandraThota) [reply examples/basic/BasicComplex/BasicSendComplex.go]: Changed back to the main repo

File: examples/basic/BasicFromHtmlFile/BasicFromHtmlFile.go (hunk 3 above)
Comment (PraneethChandraThota) [reply examples/basic/BasicFromHtmlFile/BasicFromHtmlFile.go]: Changed back to the main repo

File: examples/basic/BasicSendWithProxy/BasicSendWithProxy.go (hunk 4 above)
Comment (PraneethChandraThota) [reply examples/basic/BasicSendWithProxy/BasicSendWithProxy.go]: Changed back to the main repo

File: examples/basic/BasicWithAmpBody/BasicWithAmpBody.go (hunk 5 above)
Comment (PraneethChandraThota) [reply examples/basic/BasicWithAmpBody/BasicWithAmpBody.go]: Changed back to the main repo

File: injectionapi/core/injectionrequestfactory.go (hunk 22 above)
Comment (PraneethChandraThota) [reply injectionapi/core/injectionrequestfactory.go]: Changed back to the main repo

File: injectionapi/sendvalidator.go (hunk 23 above)
Comment (PraneethChandraThota) [reply injectionapi/sendvalidator.go]: Changed back to the main repo

File: injectionapi/socketlabsclient.go (hunk 24 above)
Comment (PraneethChandraThota) [reply injectionapi/socketlabsclient.go]: Changed back to the main repo

File: injectionapi/socketlabsclient.go (hunk 25 above)
Comment (PraneethChandraThota) [reply injectionapi/socketlabsclient.go]: added const

File: injectionapi/socketlabsclient.go (hunk 26 above)
Comment (PraneethChandraThota) [reply injectionapi/socketlabsclient.go]: added const

**Review by david-schrenker — APPROVED** [review]

**Pull request merged**